File: programmers/javascript/level2/오픈채팅방.js
const record = [
  "Enter uid1234 Muzi",
  "Enter uid4567 Prodo",
  "Leave uid1234",
  "Enter uid1234 Prodo",
  "Change uid4567 Ryan",
];

const solution = (record) => {
  let answer = [];

  let newArr = record.map((str) => str.split(" "));

  //최종 닉네임으로 변경
  let nickNameSet = {};
  for (let i = 0; i < newArr.length; i++) {
    if (newArr[i][0] === "Enter" || newArr[i][0] === "Change") {
      nickNameSet[newArr[i][1]] = newArr[i][2];
    }
  }

  //출력
  for (let i = 0; i < newArr.length; i++) {
    if (newArr[i][0] === "Enter") {
      answer.push(nickNameSet[newArr[i][1]] + "님이 들어왔습니다.");
    } else if (newArr[i][0] === "Leave") {
      answer.push(nickNameSet[newArr[i][1]] + "님이 나갔습니다.");
    }
  }
  return answer;
};

console.log(solution(record));

File: python/programmers/level2/오픈채팅방.py
def solution(records):
    answer = []
    dic = {}
    # 최종 닉네임을 결정
    for record in records:
        com = record.split(" ")
        if com[0] == 'Enter':
            dic[com[1]] = com[2]
        elif com[0] == 'Change':
            dic[com[1]] = com[2]

    # 답에 추가
    for record in records:
        com = record.split(" ")
        if com[0] == 'Enter':
            answer.append(dic[com[1]] + "님이 들어왔습니다.")
        elif com[0] == 'Leave':
            answer.append(dic[com[1]] + "님이 나갔습니다.")

    return answer


print(solution(
    ["Enter uid1234 Muzi", "Enter uid4567 Prodo", "Leave uid1234", "Enter uid1234 Prodo", "Change uid4567 Ryan"]))
